src/styles/utils/_loops.scss

```diff
@@ -0,0 +1,17 @@
+@for $index from 1 through $calendarMonthDays {
+  .calendar__item:nth-child(#{$index})::before {
+    content: "#{$index}";
+  }
+}
+
+@each $day, $margin in $days {
+  .calendar--start-day--#{$day} :first-child {
+    margin-left: $margin;
+  }
+}
+
+@for $value from 28 through $calendarMonthDays {
+  .calendar--month-length--#{$value} :nth-child(n + #{$value + 1}) {
+    display: none;
+  }
+}
```

Comment:
You don't need to move loops into separate file, keep it with corresponding block.